elyra/pipeline/component.py

```diff
@@ -141,11 +147,10 @@ def __init__(self, id: str, name: str, description: Optional[str], source_type:
         self._description = description
         self._source_type = source_type
         self._source = source
-        self._catalog_entry_id = catalog_entry_id
 
         self._runtime = runtime
         self._op = op
-        self._category_id = category_id
+        self._category_ids = categories
```

Comment:
We can remove the conditional expression in the `categories` property handler via an `or` clause.
```suggestion
        self._category_ids = categories or []
```